streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -785,6 +785,11 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
         // consumer/producer configurations, log a warning and remove the user defined value from the Map.
         // Thus the default values for these consumer/producer configurations that are suitable for
         // Streams will be used instead.
+        final Object maxInflightRequests = clientProvidedProps.get(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION);
+        if (eosEnabled && maxInflightRequests != null && 5 < (int) maxInflightRequests) {
+            throw new ConfigException("Must set " + ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION + " to at most 5" +
+                    " to use the idempotent producer.");
```

Comment:
nit: for exception message `ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION  + " can't exceed 5 when using the idempotent producer"`

Reply:
Done